src/button-dropdown/index.js

```diff
@@ -34,76 +35,66 @@ class ButtonDropdown extends Component {
   }
 
   state = {
-    isDropdownOpen: false
-  }
-
-  _getButtonClasses() {
-    var classes = this.props.type;
-    if (this.props.title.trim().length > 0)
-      classes += ' button-with-text';
-    return classes;
-  }
-
-  _renderTitle() {
-    if (this.props.title.trim().length > 0)
-      return <span ref='title'>{this.props.title}</span>
-    else {
-      return <span ref='title'>&nbsp;</span>
-    }
+    isDropdownOpen: false,
   }
 
   render() {
-    const { style, disabled, type, title } = this.props;
-    return <div style={style} className={'button-dropdown ' + this._isDropdownOpen()} disabled={disabled} ref='buttonDropdown'>
-      <button className={this._getButtonClasses()} onClick={this._toggleDropdownOptions} ref='button'>
-        {this._renderTitle()}
-        <span className='arrow ic ic-chevron-down' ref='icon'></span>
-      </button>
-      <ul className={this._getOptionsClasses()} ref='options'>
-        {this._buildDropdown()}
-      </ul>
-    </div>;
-  }
+    const { style, disabled, type, title, children } = this.props;
+
+    const buttonDropClasses = this.state.isDropdownOpen
+      ? 'button-dropdown show'
+      : 'button-dropdown';
+
+    const dropdown = disabled || !this.state.isDropdownOpen
```

Comment:
guarded method?
`methodName() {
  if (dropdown = disabled || !this.state.isDropdownOpen) return null;
  return <dropdownstuff/>;
}
`